src/plan_expand_hypertable.c

```diff
@@ -51,54 +92,67 @@ collect_quals_walker(Node *node, CollectQualCtx *ctx)
 		{
 			Node	   *qual = (Node *) lfirst(lc);
 			RestrictInfo *restrictinfo;
+			Relids		relids;
+			bool		belongs_to_cur_rel = false;
 
-			Relids		relids = pull_varnos(qual);
+			relids = pull_varnos(qual);
 
-			if (bms_num_members(relids) != 1 || !bms_is_member(ctx->rel->relid, relids))
+			belongs_to_cur_rel = bms_num_members(relids) == 1 && bms_is_member(ctx->rel->relid, relids);
+
+			if (!belongs_to_cur_rel)
 				continue;
+
+			if (is_chunk_exclusion_func(qual))
+			{
+				FuncExpr   *func_expr = (FuncExpr *) qual;
+
+				/* validation */
+				if (ctx->chunk_exclusion_func != NULL && belongs_to_cur_rel)
+					ereport(ERROR,
+							(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+							 errmsg("only one chunks_in call is allowed per hypertable")));
+
+				Assert(func_expr->args->length == 2);
```

Comment:
Well, as you said if we got here then this is indeed the right chunk exclusion function so that's not what I was aiming for. Since we'll be explicitly accessing first and second list element I think that it's a good practice to Assert that list has size 2.